### PR TITLE
Added Josephus Problem in Every Language Article

### PR DESCRIPTION
**Pull request by AakashBelide**
This is in reference to the enhancement for PR in TheRenegadeCoder/sample-programs#1933

### CODE REVIEW

File: projects/josephus-problem/index.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+Some tests for your program are:
+
+
+| Description | Input (n, k) | Output |
+| :---------- | :---- | :----- |
+| No Input                    |      | "Usage: please input the total number of people and number of people to skip." |
+| Empty Input                 | ""   | "Usage: please input the total number of people and number of people to skip." |
+| Invalid Input: not a number | word | "Usage: please input the total number of people and number of people to skip." |
+| Sample Input: 0, 1 (0 people in circle)  | 0, 1  | "Usage: please input the total number of people and number of people to skip." |
```
Comment (jrg94) [inline projects/josephus-problem/index.md:33]:
```suggestion
| Sample Input: 0, 1 (0 people in circle)  | 0 1  | "Usage: please input the total number of people and number of people to skip." |
```

I think for simplicity you can leave the comma out. Perhaps these should even be separate columns. That way, no one has to parse the comma on the command line.

File: projects/josephus-problem/index.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+| Sample Input: 5, 2  | 5, 2  | 3 |
+| Sample Input: 7, 3  | 7, 3  | 4 |
+| Sample Input: 41, 4  | 41, 4  | 11 |
+
+## Articles
+
+{% include article_list.md collection=site.categories.[name of project] % }
```
Comment (jrg94) [inline projects/josephus-problem/index.md:40]:
```suggestion
{% include article_list.md collection=site.categories.josephus-problem %}
```

File: projects/josephus-problem/index.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+---
+title: Josephus Porblem in Every Language
+layout: default
+date: 2020-10-03
+last-modified: 2020-10-03
+featured-image: 
+tags: [recursion]
```
Comment (jrg94) [inline projects/josephus-problem/index.md:7]:
```suggestion
tags: [josephus-problem]
```

File: projects/josephus-problem/index.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+---
+title: Josephus Porblem in Every Language
```
Comment (jrg94) [inline projects/josephus-problem/index.md:2]:
```suggestion
title: Josephus Problem in Every Language

```

File: projects/josephus-problem/index.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+---
+title: Josephus Porblem in Every Language
+layout: default
+date: 2020-10-03
+last-modified: 2020-10-03
+featured-image: 
+tags: [recursion]
+authors:
+---
```
Comment (jrg94) [inline projects/josephus-problem/index.md:9]: Make sure to add yourself as an author! You'll need to put yourself in the authors.yml file as well.

**Review by jrg94 — CHANGES_REQUESTED** [review]
Overall, really nice work! Just a couple minor changes.

File: projects/josephus-problem/index.md (hunk 5 above)
Comment (AakashBelide) [reply projects/josephus-problem/index.md:11]: Done! Thank you

File: projects/josephus-problem/index.md (hunk 1 above)
Comment (AakashBelide) [reply projects/josephus-problem/index.md]: I have added a separate column. Thanks for the suggestions!

File: projects/josephus-problem/index.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
+---
+title: Josephus Problem in Every Language
+
+layout: default
+date: 2020-10-03
```
Comment (jrg94) [inline projects/josephus-problem/index.md:5]:
```suggestion
date: 2020-10-06
```

File: projects/josephus-problem/index.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
+---
+title: Josephus Problem in Every Language
+
+layout: default
+date: 2020-10-03
+last-modified: 2020-10-03
```
Comment (jrg94) [inline projects/josephus-problem/index.md:6]:
```suggestion
last-modified: 2020-10-06
```

**Review by jrg94 — CHANGES_REQUESTED** [review]
Awesome! I think we can update the dates and we'll be good to go.

**Comment by jrg94** [comment]
Actually, when you say "index", do you mean starting from 1 or starting from 0? That should probably be documented in here.

**Comment by AakashBelide** [comment]
> 
> 
> Actually, when you say "index", do you mean starting from 1 or starting from 0? That should probably be documented in here.

It's starting from 0. Here is a demonstartion of how it works:
(Input: 5(total number of people in circle, 2 (number of people to skip -1)
Output: (Number of people in the circle currently, number of people to skip, index of the person to be killed or removed)
[1, 2, 3, 4, 5] 1 0
[1, 3, 4, 5] 1 1
[1, 3, 5] 1 2
[3, 5] 1 0
3
Hence, the 3rd person remains alive at the end.
Should I add this as an example and mention it in the description of index.md?

**Comment by jrg94** [comment]
Is there any way you could add this example to the description? I think it would help a lot!

**Comment by AakashBelide** [comment]
> 
> 
> Is there any way you could add this example to the description? I think it would help a lot!

Shall I add it as an example in the description or shall I make a new example sub heading and add under it?

**Comment by jrg94** [comment]
Good question! I think we usually put examples as a sub heading under Description. See Bubble Sort: https://sample-programs.therenegadecoder.com/projects/bubble-sort/

**Comment by AakashBelide** [comment]
> 
> 
> Good question! I think we usually put examples as a sub heading under Description. See Bubble Sort: https://sample-programs.therenegadecoder.com/projects/bubble-sort/

I have added an example. Kindly check it and let me know if there are any more changes to be made.

File: projects/josephus-problem/index.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+### Example
+
+__Input:__
+
+n = 5 (total number of people in circle)
+
+k = 2 (number of people - 1 are skipped and kth person is killed)
+
+Output: (Number of people in the circle currently, number of people to skip, index of the person to be killed or removed)
```
Comment (jrg94) [inline projects/josephus-problem/index.md:27]:
```suggestion
__Output__: (Number of people in the circle currently, number of people to skip, index of the person to be killed or removed)
```

**Review by jrg94 — APPROVED** [review]
Looks great! Just a quick fix and we're ready to go

**Comment by AakashBelide** [comment]
> 
> 
> Looks great! Just a quick fix and we're ready to go

Great!!

**Pull request merged**